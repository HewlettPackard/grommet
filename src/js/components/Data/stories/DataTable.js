import React from 'react';

import {
  DataFilters,
  DataFilter,
  DataSearch,
  DataSort,
  DataSummary,
  DataTable,
  Grid,
  Notification,
  Toolbar,
} from 'grommet';

import { Data } from '../Data';
import { columns, DATA } from '../../DataTable/stories/data';

export const Table = () => (
  // Uncomment <Grommet> lines when using outside of storybook
  // <Grommet theme={...}>
  <Grid
    flex={false}
    pad="large"
    columns={[['small', 'large']]}
    justifyContent="center"
    gap="large"
  >
    <Notification
      status="info"
      message="Data is in 'beta'. The API surface is subject to change."
    />
    <Data data={DATA}>
      <Toolbar>
        <DataSearch />
        <DataFilters drop>
          <DataFilter property="location" />
          <DataSort />
        </DataFilters>
      </Toolbar>
      <DataSummary />
      <DataTable columns={columns} />
    </Data>
  </Grid>
  // </Grommet>
);

Table.storyName = 'DataTable';

Table.args = {
  full: true,
};

export default {
  title: 'Layout/Data/DataTable',
};
