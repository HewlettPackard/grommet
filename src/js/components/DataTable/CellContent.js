import React from 'react';

import { Text } from '../Text';

const CellContent = ({ datum, property, render, primary }) => {
  let content = render ? render(datum) : datum[property];
  if (typeof content === 'string' || typeof content === 'number') {
    if (primary) {
      content = <strong>{content}</strong>;
    }
    content = <Text>{content}</Text>;
  }
  return content;
};

export default CellContent;
