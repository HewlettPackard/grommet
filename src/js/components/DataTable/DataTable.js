import React, { Component } from 'react';

import { Header } from './Header';
import { Footer } from './Footer';
import { Body } from './Body';
import { GroupedBody } from './GroupedBody';
import { buildState } from './buildState';
import { StyledDataTable } from './StyledDataTable';

const contexts = ['header', 'body', 'footer'];

const normalizeProp = (prop, context) => {
  if (prop) {
    if (prop[context]) return prop[context];
    if (contexts.some(c => prop[c])) return undefined;
    return prop;
  }
  return undefined;
};

class DataTable extends Component {
  static defaultProps = {
    columns: [],
    data: [],
    step: 50,
  };

  state = {};

  static getDerivedStateFromProps(nextProps, prevState) {
    return buildState(nextProps, prevState);
  }

  onFiltering = property => {
    this.setState({ filtering: property });
  };

  onFilter = (property, value) => {
    /* eslint-disable-next-line react/prop-types */
    const { onSearch } = this.props;
    const { filters } = this.state;
    const nextFilters = { ...filters };
    nextFilters[property] = value;
    this.setState({ filters: nextFilters });

    // Let caller know about search, if interested
    if (onSearch) {
      onSearch(nextFilters);
    }
  };

  onSort = property => () => {
    const { sort } = this.state;
    const ascending =
      sort && property === sort.property ? !sort.ascending : true;
    this.setState({ sort: { property, ascending } });
  };

  onToggleGroup = groupValue => () => {
    const { groupState } = this.state;
    const { groupBy } = this.props;
    const nextGroupState = { ...groupState };
    nextGroupState[groupValue] = {
      ...nextGroupState[groupValue],
      expanded: !nextGroupState[groupValue].expanded,
    };
    this.setState({ groupState: nextGroupState });
    if (groupBy.onExpand) {
      const expandedKeys = Object.keys(nextGroupState).filter(
        k => nextGroupState[k].expanded,
      );
      groupBy.onExpand(expandedKeys);
    }
  };

  onToggleGroups = () => {
    const { groupState } = this.state;
    const { groupBy } = this.props;
    const expanded =
      Object.keys(groupState).filter(k => !groupState[k].expanded).length === 0;
    const nextGroupState = {};
    Object.keys(groupState).forEach(k => {
      nextGroupState[k] = { ...groupState[k], expanded: !expanded };
    });
    this.setState({ groupState: nextGroupState });
    if (groupBy.onExpand) {
      const expandedKeys = Object.keys(nextGroupState).filter(
        k => nextGroupState[k].expanded,
      );
      groupBy.onExpand(expandedKeys);
    }
  };

  onResize = property => width => {
    const { widths } = this.state;
    const nextWidths = { ...(widths || {}) };
    nextWidths[property] = width;
    this.setState({ widths: nextWidths });
  };

  render() {
    const {
      background,
      border,
      /* eslint-disable-next-line react/prop-types */
      columns,
      data: propsData,
      groupBy,
      onMore,
      replace,
      pad,
      resizeable,
      rowProps,
      size,
      sortable,
      step,
      onClickRow, // removing unknown DOM attributes
      onSearch, // removing unknown DOM attributes
      ...rest
    } = this.props;
    const {
      data,
      filtering,
      filters,
      footerValues,
      groups,
      groupState,
      primaryProperty,
      showFooter,
      sort,
      widths,
    } = this.state;

    if (size && resizeable) {
      console.warn('DataTable cannot combine "size" and "resizeble".');
    }

    return (
      <StyledDataTable {...rest}>
        <Header
          background={normalizeProp(background, 'header')}
          border={normalizeProp(border, 'header')}
          columns={columns}
          filtering={filtering}
          filters={filters}
          groups={groups}
          groupState={groupState}
          pad={normalizeProp(pad, 'header')}
          size={size}
          sort={sort}
          widths={widths}
          onFiltering={this.onFiltering}
          onFilter={this.onFilter}
          onResize={resizeable ? this.onResize : undefined}
          onSort={sortable ? this.onSort : undefined}
          onToggle={this.onToggleGroups}
        />
        {groups ? (
          <GroupedBody
            background={normalizeProp(background, 'body')}
            border={normalizeProp(border, 'body')}
            columns={columns}
            groupBy={groupBy.property ? groupBy.property : groupBy}
            groups={groups}
            groupState={groupState}
            pad={normalizeProp(pad, 'body')}
            primaryProperty={primaryProperty}
            onToggle={this.onToggleGroup}
          />
        ) : (
          <Body
            background={normalizeProp(background, 'body')}
            border={normalizeProp(border, 'body')}
            columns={columns}
            data={data}
            onMore={onMore}
            replace={replace}
            onClickRow={onClickRow}
            pad={normalizeProp(pad, 'body')}
            primaryProperty={primaryProperty}
            rowProps={rowProps}
            size={size}
            step={step}
          />
        )}
        {showFooter && (
          <Footer
            background={normalizeProp(background, 'footer')}
            border={normalizeProp(border, 'footer')}
            columns={columns}
            footerValues={footerValues}
            groups={groups}
            pad={normalizeProp(pad, 'footer')}
            primaryProperty={primaryProperty}
            size={size}
          />
        )}
      </StyledDataTable>
    );
  }
}

let DataTableDoc;
if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  DataTableDoc = require('./doc').doc(DataTable);
}
const DataTableWrapper = DataTableDoc || DataTable;

export { DataTableWrapper as DataTable };
