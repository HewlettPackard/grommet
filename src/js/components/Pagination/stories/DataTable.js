import React from 'react';
import { storiesOf } from '@storybook/react';

import { Box, Grommet, DataTable } from 'grommet';
// import { grommet } from 'grommet/themes';
import { hpe } from 'grommet-theme-hpe';
import { columns, DATA } from '../../DataTable/stories/data';

const PaginatedDataTable = () => {
  const [select, setSelect] = React.useState([]);
  return (
    <Grommet theme={hpe}>
      <Box
        background="background-back"
        margin="large"
        pad="small"
        round="small"
      >
        <DataTable
          columns={columns}
          data={[...DATA]}
          step={10}
          select={select}
          onSelect={setSelect}
          paginationProps={{
            margin: { top: 'small' },
            pad: { vertical: 'small' },
            round: 'xsmall',
            step: 4,
          }}
          sortable
        />
      </Box>
      <Box
        background="background-back"
        margin="large"
        pad="small"
        round="small"
      >
        <DataTable
          columns={columns}
          data={[...DATA]}
          step={10}
          select={select}
          onSelect={setSelect}
          paginationProps={{
            // alignSelf: 'center',
            background: 'background-contrast',
            justify: 'center',
            fill: true,
            margin: { top: 'medium' },
            pad: { vertical: 'small' },
            round: 'xsmall',
            step: 7,
          }}
          sortable
        />
      </Box>
      <Box
        background="background-back"
        margin="large"
        pad="small"
        round="small"
      >
        <DataTable
          columns={columns}
          data={[...DATA]}
          step={10}
          select={select}
          onSelect={setSelect}
          sortable
          pin
        />
      </Box>
    </Grommet>
  );
};

storiesOf('Pagination', module).add('DataTable', () => <PaginatedDataTable />);
