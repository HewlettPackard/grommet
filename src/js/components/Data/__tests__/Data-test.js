import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Grommet } from '../../Grommet';
import { DataTable } from '../../DataTable';
import { Data } from '..';

const data = [{ name: 'a' }, { name: 'b' }];

describe('Data', () => {
  test('renders', () => {
    const { container } = render(
      <Grommet>
        <Data data={data} />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('properties', () => {
    const { container } = render(
      <Grommet>
        <Data data={data} properties={{ name: { label: 'Name' } }} />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('toolbar', () => {
    const { container } = render(
      <Grommet>
        <Data data={data} toolbar />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('view', () => {
    const { container } = render(
      <Grommet>
        <Data data={data} view={{ search: '', properties: {} }} />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('all', () => {
    const { container } = render(
      <Grommet>
        <Data
          data={data}
          properties={{ name: { label: 'Name' } }}
          view={{ search: 'a', properties: {} }}
          toolbar
        >
          <DataTable />
        </Data>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
