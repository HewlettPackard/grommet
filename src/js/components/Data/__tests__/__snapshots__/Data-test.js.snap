// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data all 1`] = `
.c6 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c6 g {
  fill: inherit;
  stroke: inherit;
}

.c6 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c6 *[stroke*="#"],
.c6 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c6 *[fill-rule],
.c6 *[FILL-RULE],
.c6 *[fill*="#"],
.c6 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #7D4CDB;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 24px;
  min-width: 24px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 24px;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c14 {
  font-size: 14px;
  line-height: 20px;
  color: #FFFFFF;
  font-weight: normal;
}

.c16 {
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c24 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c10 {
  position: relative;
}

.c11 {
  position: relative;
  display: block;
}

.c12 {
  position: absolute;
  top: 0;
  right: 0;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c9:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus > circle,
.c9:focus > ellipse,
.c9:focus > line,
.c9:focus > path,
.c9:focus > polygon,
.c9:focus > polyline,
.c9:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus::-moz-focus-inner {
  border: 0;
}

.c9:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible) > circle,
.c9:focus:not(:focus-visible) > ellipse,
.c9:focus:not(:focus-visible) > line,
.c9:focus:not(:focus-visible) > path,
.c9:focus:not(:focus-visible) > polygon,
.c9:focus:not(:focus-visible) > polyline,
.c9:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c15 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c15:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c15:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c15:focus > circle,
.c15:focus > ellipse,
.c15:focus > line,
.c15:focus > path,
.c15:focus > polygon,
.c15:focus > polyline,
.c15:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c15:focus::-moz-focus-inner {
  border: 0;
}

.c15:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c15:focus:not(:focus-visible) > circle,
.c15:focus:not(:focus-visible) > ellipse,
.c15:focus:not(:focus-visible) > line,
.c15:focus:not(:focus-visible) > path,
.c15:focus:not(:focus-visible) > polygon,
.c15:focus:not(:focus-visible) > polyline,
.c15:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c15:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c19 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c22 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c17 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c7 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  padding-left: 48px;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus > circle,
.c7:focus > ellipse,
.c7:focus > line,
.c7:focus > path,
.c7:focus > polygon,
.c7:focus > polyline,
.c7:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus::-moz-focus-inner {
  border: 0;
}

.c7::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c7::-moz-placeholder {
  color: #AAAAAA;
}

.c7:-ms-input-placeholder {
  color: #AAAAAA;
}

.c7::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c7::-moz-focus-inner {
  border: none;
  outline: none;
}

.c7:-moz-placeholder,
.c7::-moz-placeholder {
  opacity: 1;
}

.c4 {
  position: relative;
  width: 100%;
}

.c5 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-packjustify: center;
  -webkit-justify: center;
  -ms-flex-packjustify: center;
  justify: center;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
  left: 12px;
}

.c18 {
  position: relative;
  border-spacing: 0;
  border-collapse: separate;
  height: 100%;
}

.c21:focus {
  outline: 2px solid #6FFFB0;
}

.c21:focus:not(:focus-visible) {
  outline: none;
}

@media only screen and (max-width:768px) {
  .c13 {
    border-radius: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    width: 6px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <form>
          <div
            class="c1"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
              >
                <svg
                  aria-label="Search"
                  class="c6"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m15 15 7 7-7-7zm-5.5 2a7.5 7.5 0 1 0 0-15 7.5 7.5 0 0 0 0 15z"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <input
                autocomplete="off"
                class="c7"
                name="_search"
                value="a"
              />
            </div>
          </div>
        </form>
        <div
          class="c8"
        />
        <div
          class="c3"
        >
          <button
            aria-label="Open Drop"
            class="c9"
            type="button"
          >
            <div
              class="c10"
            >
              <div
                class="c11"
              >
                <svg
                  aria-label="Filter"
                  class="c6"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m3 6 7 7v8h4v-8l7-7V3H3z"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="c12"
                style="top: -12px; right: -12px;"
              >
                <div
                  class="c13 Badge__StyledBadgeContainer-sc-1es4ws1-0"
                >
                  <span
                    class="c14"
                  >
                    1
                  </span>
                </div>
              </div>
            </div>
          </button>
          <div
            class="c8"
          />
          <div
            class="c1"
          >
            <button
              class="c15"
              type="button"
            >
              Clear filters
            </button>
          </div>
        </div>
      </div>
    </div>
    <span
      class="c16"
    >
      1 results of 2 items
    </span>
    <table
      class="c17 c18"
    >
      <thead
        class="StyledTable__StyledTableHeader-sc-1m3u5g-4 StyledDataTable__StyledDataTableHeader-sc-xrlyjm-4"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2 "
        >
          <th
            class="c19 "
            scope="col"
          >
            <div
              class="c20"
            />
          </th>
        </tr>
      </thead>
      <tbody
        class="StyledTable__StyledTableBody-sc-1m3u5g-3 c21"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2 "
        >
          <th
            class="c22 "
            scope="row"
          >
            <div
              class="c23"
            >
              <span
                class="c24"
              >
                a
              </span>
            </div>
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Data properties 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;

exports[`Data renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;

exports[`Data toolbar 1`] = `
.c6 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c6 g {
  fill: inherit;
  stroke: inherit;
}

.c6 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c6 *[stroke*="#"],
.c6 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c6 *[fill-rule],
.c6 *[FILL-RULE],
.c6 *[fill*="#"],
.c6 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c10 {
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c9 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c9:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus > circle,
.c9:focus > ellipse,
.c9:focus > line,
.c9:focus > path,
.c9:focus > polygon,
.c9:focus > polyline,
.c9:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c9:focus::-moz-focus-inner {
  border: 0;
}

.c9:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible) > circle,
.c9:focus:not(:focus-visible) > ellipse,
.c9:focus:not(:focus-visible) > line,
.c9:focus:not(:focus-visible) > path,
.c9:focus:not(:focus-visible) > polygon,
.c9:focus:not(:focus-visible) > polyline,
.c9:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c9:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c7 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  padding-left: 48px;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus > circle,
.c7:focus > ellipse,
.c7:focus > line,
.c7:focus > path,
.c7:focus > polygon,
.c7:focus > polyline,
.c7:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c7:focus::-moz-focus-inner {
  border: 0;
}

.c7::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c7::-moz-placeholder {
  color: #AAAAAA;
}

.c7:-ms-input-placeholder {
  color: #AAAAAA;
}

.c7::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c7::-moz-focus-inner {
  border: none;
  outline: none;
}

.c7:-moz-placeholder,
.c7::-moz-placeholder {
  opacity: 1;
}

.c4 {
  position: relative;
  width: 100%;
}

.c5 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-packjustify: center;
  -webkit-justify: center;
  -ms-flex-packjustify: center;
  justify: center;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  pointer-events: none;
  left: 12px;
}

@media only screen and (max-width:768px) {
  .c8 {
    width: 6px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <form>
          <div
            class="c1"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
              >
                <svg
                  aria-label="Search"
                  class="c6"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m15 15 7 7-7-7zm-5.5 2a7.5 7.5 0 1 0 0-15 7.5 7.5 0 0 0 0 15z"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <input
                autocomplete="off"
                class="c7"
                name="_search"
                value=""
              />
            </div>
          </div>
        </form>
        <div
          class="c8"
        />
        <div
          class="c3"
        >
          <button
            aria-label="Open Drop"
            class="c9"
            type="button"
          >
            <svg
              aria-label="Filter"
              class="c6"
              viewBox="0 0 24 24"
            >
              <path
                d="m3 6 7 7v8h4v-8l7-7V3H3z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <span
      class="c10"
    >
      2 items
    </span>
  </div>
</div>
`;

exports[`Data view 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;
